superbench/common/utils/network.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Network Utility."""
+
+import socket
+import subprocess
+
+
+def get_free_port():
+    """Get a free port in current system.
+
+    Return:
+        port (int): a free port in current system.
+    """
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        s.bind(('', 0))
+        s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        return s.getsockname()[1]
+    except OSError:
+        return None
+    finally:
+        s.close()
+
+
+def get_ib_devices():
+    """Get available IB devices in the system and filter ethernet devices.
+
+    Return:
+        ib_devices (list): IB devices in current system.
+    """
+    command_get_devices = 'ls /sys/class/infiniband/'
+    output = subprocess.run(
+        command_get_devices,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        shell=True,
+        check=False,
+        universal_newlines=True
+    )
+    devices = output.stdout.split()
+    devices.sort()
```

Comment:
use natural sort order? e.g., what if there are 16 devices mlx5_0, mlx5_1, ..., mlx5_15
```suggestion
    devices.sort(key=lambda s: [int(ch) if ch.isdigit() else ch for ch in re.split(r'(\d+)', s)])
```

Reply:
thanks for the comment